python/dgl/distributed/rpc_server.py

```diff
@@ -78,7 +79,9 @@ def start_server(server_id, ip_config, num_servers, num_clients, server_state, \
         client_namebook[client_id] = addr
     for client_id, addr in client_namebook.items():
         client_ip, client_port = addr.split(':')
-        assert rpc.connect_receiver(client_ip, client_port, client_id)
+        # TODO[Rhett]: server should not be blocked endlessly.
+        while not rpc.connect_receiver(client_ip, client_port, client_id):
+            time.sleep(1)
```

Comment:
why do we need to do this?

Reply:
sometimes `rpc_client::receiver` is not waiting for connection, so we need to re-try...